api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/ClusterEndpointConfigProvider.java

```diff
@@ -63,30 +54,30 @@ default Optional<String> getBindAddress() {
     }
 
     /**
-     * Indicates if this provider requires exclusive use of ports that it defines.  If true,
-     * no other provider will be able to bind the same port.
+     * Indicates if this provider requires the use of TLS.
      *
-     * @return true if this endpoint provider requires exclusive use of ports.
+     * @return true if this provider requires the use of TLS.
      */
-    default boolean requiresPortExclusivity() {
-        return true;
+    default boolean requiresTls() {
+        return false;
     }
 
     /**
-     * Indicates if this provider requires the use of TLS.
+     * Set of ports number that this provider requires exclusive use.
      *
-     * @return true if this provider requires the use of TLS.
+     * @return set of port numbers
      */
-    default boolean requiresTls() {
-        return false;
+    default Set<Integer> getExclusivePorts() {
+        return Set.of();
     }
```

Comment:
In the SNI case, virtual clusters can share a single port if they are so configured.
In the port-per-broker case, every port is exclusive.

Reply:
Yes I understand that but I don't follow why that surfaces into the API. Every port can only be bound once so it has to be unique regardless of what happens to the traffic received over it. 

The bit I'm not following is why we need to expose two different sets via the API (I can sort of rationalise implementations maintaining independent sets for validation purposes).

Reply:
>  why we need to expose two different sets via the API

I've got virtual cluster A that is prepared to shared port 5050 as it uses CECP `SniRouter`. virtual cluster B is also prepared to share 5050 as it uses SNI too.  virtual cluster C uses CECP `BrokerPerPort` which is configured to use port range 5040-5050.  We need some way to determine that A/B can happily co-exist whereas A/C and B/C can't.  That is how I arrived at the idea of the the  CECP implementations announcing what ports they might use and whether they can share or not.  Maybe there's a better way.
 
* I'm planning the existing CECP `StaticCluster` will evolve into `BrokerPerPort`. It will be configured with a port range from which the cluster's brokers will be accessed.